src/Components/Web.Extensions/src/wwwroot/headManager.js

```diff
@@ -0,0 +1,51 @@
+// Local helpers
+
+const getHeadElement = (tagName, id) => {
+    const elements = Array.from(document.getElementsByTagName(tagName));
+    return elements.find(e => e._blazorId === id);
+}
+
+const getOrCreateHeadElement = (tagName, id) => {
+    let element = getHeadElement(tagName, id);
+
+    if (element) {
+        return element;
+    }
+
+    element = document.createElement(tagName);
+    element._blazorId = id;
+
+    const head = document.getElementsByTagName('head')[0];
+    head.appendChild(element);
+
+    return element;
+};
+
+// Exported functions
+
+const setTitle = (title) => {
+    document.title = title;
+};
+
+const setTag = (tagName, id, attributes) => {
+    let tag = getOrCreateHeadElement(tagName, id);
+
+    if (attributes) {
+        for (let key in attributes) {
+            if (attributes.hasOwnProperty(key)) {
+                tag.setAttribute(key, attributes[key]);
+            }
+        }
+    }
```

Comment:
What if the developer is trying to remove an attribute? You could either explicitly remove each old attribute that's no longer in the set, or you could just destroy the old element completely and replace it with a new one.